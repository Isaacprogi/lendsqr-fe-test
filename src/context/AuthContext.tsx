import * as React from 'react'
import { AuthContextType } from '../utils/types/interface_and_types';




export const AuthContext = React.createContext<AuthContextType | null>(null)

export const AuthProvider = ({children}:{children:any}) => {
 const [auth,setAuth] = React.useState<boolean>(false)
 
 const login =() =>{
      setAuth(true)
 }
 const logout = ()=> setAuth(false)

    return <AuthContext.Provider value={{auth,login,logout}}>
               {children}
         </AuthContext.Provider>
}
