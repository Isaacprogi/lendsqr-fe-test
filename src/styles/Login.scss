@import '../styles/variables/mixins';

.log-in-page {
    width: 100%;
    height: 100%;
    // background-color: red;

    section.logo-section{
        display: none;      

        @include mediumDevices {
            display: flex;
            align-items: center;
            justify-content: center;      
        }

        img{
            width: 100%;
            max-width: 600px;
        }

        .logo{
            display: flex;
            align-items: center;
            justify-content: flex-start;
            position: absolute;
            top: 50px;
            left:92px;

            img.union{
                width: 24.75px;
                height: 25px;
            }
            img.lendsqr{
                margin-top: .4rem;
                width: 138px;
                height: 30px;
            }

        }

    }
    section.form-section{
        width: 100%;
        height: 100%;
        position: relative;
        padding: 1rem;
        position: relative;
        max-width: 40rem;
        margin: 0 auto;

        .error{
            height: 3rem;
            margin: .3rem;
            color: rgb(189, 68, 68);
        }

        .logo{
            display: flex;
            align-items: center;
            justify-content: center;

            img.union{
                width: 24.75px;
                height: 25px;
            }
            img.lendsqr{
                margin-top: .4rem;
                width: 138px;
                height: 30px;
            }

            @include mediumDevices {
                display: none;
            }
        }


        .container{
            width: 100%;
            height: 100%;
            display: flex;
            align-items: flex-start;
            justify-content: center;
            flex-direction: column;
     
            h1{
                line-height: 54.64px;
                font-weight: 700;
                size: 40px;
                color: #213F7D;
                font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
            }
            
            span.login-details-note{
                margin-bottom: 1rem;
                size: 20px;
                color: $textSecondary;
                font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
                margin-bottom: 4rem;
     
                
            }
            
            form{
                display: flex;
                align-items: flex-start;
                justify-content: center;
                flex-direction: column;
                width: 100%;
                
               
                div{
                    width: 100%;
                    position: relative;
                    margin-bottom: 1rem;
                    background: $whiteColor;
                    border: 2px solid rgba(84, 95, 125, 0.15);
                    border-radius: 5px;
     
                    input{
                        width: 100%;
                        height: 50px;
                        border: none;
                        outline: none;
                        padding-inline:.2rem ;
                    }
                }

                .toggle-password {
                    position: absolute;
                    right: 10px;
                    background: none;
                    border: none;
                    cursor: pointer;
                    font-size: 1rem;
                    padding: 0;
                    margin: 0;
                    color: #39CDCC;
                    transition: background 0.3s ease, transform 0.3s ease;

                    &:hover{
                        color: rgb(99, 85, 85);

                    }
                }

                .password-input-container {
                    position: relative;
                    display: flex;
                    align-items: center;
                }
     
                a.forgot-password{
                 font-family: 'Avenir Next';
                 font-style: normal;
                 font-weight: 600;
                 font-size: 12px;
                 line-height: 16px;      
                 text-align: right;
                 letter-spacing: 0.1em;
                 text-transform: uppercase;
                 margin-bottom: 1rem;
                 color: #39CDCC;
                 cursor: pointer;
                 text-decoration: none;
                 transition: background 0.3s ease, transform 0.3s ease;

                    &:hover{
                        color: rgb(99, 85, 85);

                    }
                }
     
                button {
                    width: 100%;
                    background: #39CDCC;
                    border-radius: 8px;
                    padding: 1rem;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    outline: none;
                    cursor: pointer;
                    border: none;
                    transition: background 0.3s ease, transform 0.3s ease;
                
                    span {
                        font-family: sans-serif;
                        text-align: center;
                        color: $whiteColor;
                    }
                
                    &:hover {
                        background: darken(#39CDCC, 10%);
                    }
                }
                
            }
         }
    }

    
    

    @include mediumDevices {
        display: grid;
        grid-template-columns: 1fr 1fr;
    }
    

}