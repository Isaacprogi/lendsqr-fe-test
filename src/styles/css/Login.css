.log-in-page {
  width: 100%;
  height: 100%;
}
.log-in-page section.logo-section {
  display: none;
}
@media screen and (min-width: 768px) {
  .log-in-page section.logo-section {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
  }
}
.log-in-page section.logo-section img {
  width: 100%;
  max-width: 600px;
}
.log-in-page section.logo-section .logo {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  position: absolute;
  top: 50px;
  left: 92px;
}
.log-in-page section.logo-section .logo img.union {
  width: 24.75px;
  height: 25px;
}
.log-in-page section.logo-section .logo img.lendsqr {
  margin-top: 0.4rem;
  width: 138px;
  height: 30px;
}
.log-in-page section.form-section {
  width: 100%;
  height: 100%;
  position: relative;
  padding: 1rem;
  position: relative;
  max-width: 40rem;
  margin: 0 auto;
}
.log-in-page section.form-section .error {
  height: 3rem;
  margin: 0.3rem;
  color: rgb(189, 68, 68);
}
.log-in-page section.form-section .logo {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}
.log-in-page section.form-section .logo img.union {
  width: 24.75px;
  height: 25px;
}
.log-in-page section.form-section .logo img.lendsqr {
  margin-top: 0.4rem;
  width: 138px;
  height: 30px;
}
@media screen and (min-width: 768px) {
  .log-in-page section.form-section .logo {
    display: none;
  }
}
.log-in-page section.form-section .container {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}
.log-in-page section.form-section .container h1 {
  line-height: 54.64px;
  font-weight: 700;
  size: 40px;
  color: #213F7D;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
}
.log-in-page section.form-section .container span.login-details-note {
  margin-bottom: 1rem;
  size: 20px;
  color: #545F7D;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  margin-bottom: 4rem;
}
.log-in-page section.form-section .container form {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
}
.log-in-page section.form-section .container form div {
  width: 100%;
  position: relative;
  margin-bottom: 1rem;
  background: white;
  border: 2px solid rgba(84, 95, 125, 0.15);
  border-radius: 5px;
}
.log-in-page section.form-section .container form div input {
  width: 100%;
  height: 50px;
  border: none;
  outline: none;
  padding-inline: 0.2rem;
}
.log-in-page section.form-section .container form .toggle-password {
  position: absolute;
  right: 10px;
  background: none;
  border: none;
  cursor: pointer;
  font-size: 1rem;
  padding: 0;
  margin: 0;
  color: #39CDCC;
  -webkit-transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease, -webkit-transform 0.3s ease;
}
.log-in-page section.form-section .container form .toggle-password:hover {
  color: rgb(99, 85, 85);
}
.log-in-page section.form-section .container form .password-input-container {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}
.log-in-page section.form-section .container form a.forgot-password {
  font-family: "Avenir Next";
  font-style: normal;
  font-weight: 600;
  font-size: 12px;
  line-height: 16px;
  text-align: right;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  margin-bottom: 1rem;
  color: #39CDCC;
  cursor: pointer;
  text-decoration: none;
  -webkit-transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease, -webkit-transform 0.3s ease;
}
.log-in-page section.form-section .container form a.forgot-password:hover {
  color: rgb(99, 85, 85);
}
.log-in-page section.form-section .container form button {
  width: 100%;
  background: #39CDCC;
  border-radius: 8px;
  padding: 1rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  outline: none;
  cursor: pointer;
  border: none;
  -webkit-transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, -webkit-transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease;
  transition: background 0.3s ease, transform 0.3s ease, -webkit-transform 0.3s ease;
}
.log-in-page section.form-section .container form button span {
  font-family: sans-serif;
  text-align: center;
  color: white;
}
.log-in-page section.form-section .container form button:hover {
  background: #2ba8a8;
}
@media screen and (min-width: 768px) {
  .log-in-page {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
}/*# sourceMappingURL=Login.css.map */