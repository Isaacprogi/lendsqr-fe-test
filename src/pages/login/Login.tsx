import React, { useState, useEffect } from 'react';
import '../../styles/css/Login.css';
import PABLOSVG from '../../assets/svg/login/pablo.svg';
import UNION from '../../assets/svg/login/union.svg';
import LENDSQR from '../../assets/svg/login/lendsqr.svg';
import { useAuthContext } from '../../hooks/useAuthContext';
import { useNavigate } from 'react-router-dom';
import { Link } from 'react-router-dom';

export const Login: React.FC = () => {
    const [email, setEmail] = useState<string>('adeji@gmail.com');
    const [password, setPassword] = useState<string>('12345678');
    const [error, setError] = useState<string>('');
    const [showPassword, setShowPassword] = useState<boolean>(false);
    const { login } = useAuthContext();
    const [imageLoading, setImageLoading] = useState<boolean>(true);

    const emailCheck: string = 'adeji@gmail.com';
    const passwordCheck: string = '12345678';

    const navigate = useNavigate();

    useEffect(() => {
        const image = new Image();
        image.src = PABLOSVG;
        image.onload = () => {
            setImageLoading(false);
        };
    }, []);

    const handleSubmit = (e: React.FormEvent) => {
        e.preventDefault();
        if (email === emailCheck && password === passwordCheck) {
            login();
            return navigate('/');
        } else {
            setError('Invalid credentials');
            console.log(error);
        }
    };

    return imageLoading === false ? (
        <div className='log-in-page'>
            <section className="logo-section">
                <span className='logo'>
                    <img className='union' src={UNION} alt="logo-1" />
                    <img className='lendsqr' src={LENDSQR} alt="logo-2" />
                </span>
                <img src={PABLOSVG} alt="background-img" />
            </section>

            <section className="form-section">
                <span className='logo'>
                    <img className='union' src={UNION} alt="" />
                    <img className='lendsqr' src={LENDSQR} alt="" />
                </span>

                <div className="container">
                    <h1>Welcome!</h1>
                    <span className="login-details-note">Enter details to Login</span>

                    <form onSubmit={handleSubmit}>
                        <div>
                            <input
                                placeholder='Email'
                                type="email"
                                value={email}
                                onChange={(e) => {
                                    setEmail(e.target.value);
                                    setError("");
                                }}
                            />
                        </div>
                        <div className="password-input-container">
                            <input
                                placeholder='Password'
                                type={showPassword ? "text" : "password"}
                                value={password}
                                onChange={(e) => {
                                    setPassword(e.target.value);
                                    setError("");
                                }}
                            />
                            <span
                                className="toggle-password"
                                onClick={() => setShowPassword(!showPassword)}
                            >
                                {showPassword ? "Hide" : "Show"}
                            </span>
                        </div>
                        <Link to='#' className='forgot-password'>FORGOT PASSWORD?</Link>
                        <button disabled={(!email || !password)}>
                            <span>LOG IN</span>
                        </button>
                    </form>
                    <span className="error">
                        {error && error}
                    </span>
                </div>
            </section>
        </div>
    ) : <></>
}
