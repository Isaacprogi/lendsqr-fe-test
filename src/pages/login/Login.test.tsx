import { fireEvent, render, screen,act,waitFor} from '@testing-library/react';
import { Login } from './Login'
import { BrowserRouter } from 'react-router-dom';
import { AuthProvider } from '../../context/AuthContext';
import renderer from 'react-test-renderer'



describe('test for Login component', () => {
   const LoginComponent = async() => {
     return act(()=>{
      render(
         <AuthProvider>
            <BrowserRouter>
               <Login />
            </BrowserRouter>
         </AuthProvider>
      )
     })
   }
   beforeEach(() => {
     LoginComponent()
   })


   test('render email input', async () => {
      waitFor(()=>{
         const emailInputEl = screen.getByPlaceholderText('Email')
      expect(emailInputEl).toBeInTheDocument()
      })
   })

   test('render password input', () => {
      waitFor(()=>{
         const passwordInputEl = screen.getByPlaceholderText('Password')
      expect(passwordInputEl).toBeInTheDocument()
      })
   })

   test(' render button element', () => {
      waitFor(()=>{
         const buttonEl = screen.getByRole("button")
      expect(buttonEl).toBeInTheDocument()
      expect(buttonEl.textContent).toBe('LOG IN')
      })
   })


   test('email input should empty', () => {
     waitFor(()=>{
      const emailInputEl = screen.getByPlaceholderText("Email")
      expect(emailInputEl.textContent).toBe("")
     })
   })

   test('password input should empty', () => {
      waitFor(()=>{
         const passwordInputEl = screen.getByPlaceholderText("Password")
      expect(passwordInputEl.textContent).toBe("")
      })
   })

   test('button element should be disabled', () => {
      waitFor(()=>{
         const buttonEl = screen.getByRole("button")
      expect(buttonEl).toBeDisabled()
      })
   })

   test("email input element should  change", () => {
      waitFor(()=>{
         const emailInputEl = screen.getByPlaceholderText<HTMLInputElement>(/email/i)
      const testValue = "test"

      fireEvent.change(emailInputEl, { target: { value: testValue } })
      expect(emailInputEl.value).toBe(testValue)
      })
   })


   test("password input element should change", () => {
     waitFor(()=>{
      const passwordInputEl = screen.getByPlaceholderText<HTMLInputElement>(/password/i)
      const testValue = "test"

      fireEvent.change(passwordInputEl, { target: { value: testValue } })
      expect(passwordInputEl.value).toBe(testValue)
     })
   })

   test("button should not be disabled when inputs exist", () => {
     waitFor(()=>{
      const emailInputEl = screen.getByPlaceholderText(/email/i)
      const passwordInputEl = screen.getByPlaceholderText(/password/i)
      const buttonEl = screen.getByRole("button")
      const testValue = "check"

      fireEvent.change(emailInputEl, { target: { value: testValue } })
      fireEvent.change(passwordInputEl, { target: { value: testValue } })

      expect(buttonEl).not.toBeDisabled()
     })
   })

   test('should match snapshot', async () => {
      const tree = renderer.create(<AuthProvider>
         <BrowserRouter>
            <Login />
         </BrowserRouter>
      </AuthProvider>).toJSON();
      expect(tree).toMatchSnapshot();
   });
})























