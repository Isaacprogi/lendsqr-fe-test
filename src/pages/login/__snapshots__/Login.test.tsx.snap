// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test for Login component > should match snapshot 1`] = `null`;

exports[`test for Login component should match snapshot 1`] = `null`;
